### PR TITLE
[CI] Use `inputs` for actions to enforce correct use

### PR DESCRIPTION
**Pull request by itislu**
- [x] First merge #368 

---

- setup action:
  Making the inputs optional and checking if they are set before using them allows to set up less when not needed.
  An example is the compilation test.

- summary action:
  Using env was hard to follow and required comments in the action what was required for that action to work.

- delivery workflow:
  Split up delivery workflow into 2 jobs.
  Using jobs for controling the flow is better GitHub Actions practice than constant if checks.

### CODE REVIEW

**Review by LeaYeh — DISMISSED** [review]

**Comment by itislu** [comment]
Solved the conflicts, not significant at all. Nothing different now, no need for you to check everything again.

**Review by LeaYeh — APPROVED** [review]
